pkg/kubelet/kubelet.go

```diff
@@ -923,10 +924,12 @@ func (kl *Kubelet) listPodsFromDisk() ([]types.UID, error) {
 // Starts garbage collection threads.
 func (kl *Kubelet) StartGarbageCollection() {
 	go wait.Until(func() {
-		if err := kl.containerGC.GarbageCollect(kl.sourcesReady.AllReady()); err != nil {
-			glog.Errorf("Container garbage collection failed: %v", err)
-		}
-	}, ContainerGCPeriod, wait.NeverStop)
+		kl.gcController.startContainerGarbageCollection(func() {
```

Comment:
@ronnielai Why do we need to start the garbage collector in a `wait.Until` clause? :)